src/code/TextTableParser.cs

```diff
@@ -0,0 +1,505 @@
+            {
+                string value;
+                if (cInfos[i].Length == -1) // end of line
+                {
+                    value = line.Substring(cInfos[i].Start).Trim();
+                }
+                else
+                {
+                    value = line.Substring(cInfos[i].Start, cInfos[i].Length).Trim();
+                }
+                data[i] = value;
+
+            }
+            return data;
+        }
+
+        private string[] GetHeaderColumns(List<ColumnInfo> cInfos, string line)
+        {
+            string[] columns = new string[cInfos.Count];
+            for(int i = 0; i < cInfos.Count; i++)
+            {
+                if (cInfos[i].Length == -1) // end of line
+                {
+                    columns[i] = line.Substring(cInfos[i].Start).Trim().Replace(" ", "_");
+                }
+                else
+                {
+                    columns[i] = line.Substring(cInfos[i].Start, cInfos[i].Length).Trim().Replace(" ", "_");
+                }
+            }
+            return columns;
+        }
+
+        private int GetMaxLength(List<string>lines)
+        {
+            int maximumLength = 0;
+            foreach(string line in lines)
+            {
+                if (line.Length > maximumLength)
+                {
+                    maximumLength = line.Length;
+                }
+            }
+            return maximumLength;
+        } 
+
+        // Analyze for white space. If we find consistent white space,
+        // then we can use that to determine the columns.
+        // If the value in the array element is the same as the number of lines,
+        // we have a column.
+    
+        private int[] AnalyzeColumns(List<string>lines)
+        {
+            int maximumLength = GetMaxLength(lines);
+            int[] SpaceArray = new int[maximumLength];
+            Array.Fill(SpaceArray, 0);
+            foreach(string line in lines)
+            {
+                for(int i = 0; i < line.Length; i++)
+                {
+                    if(char.IsWhiteSpace(line[i]))
+                    {
+                        SpaceArray[i] += 1;
+                    }
+                }
+            }
+            return SpaceArray;
+        }
+
+        private List<ColumnInfo> GetColumnList(int[]StartColumns)
+        {
+            List<ColumnInfo> ColumnInfoList = new List<ColumnInfo>();
+            for(int i = 0; i < StartColumns.Length; i++)
+            {
+                int length;
+                try
+                {
+                    length = StartColumns[i+1] - StartColumns[i];
+                }
+                catch
+                {
+                    length = -1;
+                }
+                ColumnInfoList.Add(new ColumnInfo() { Start = StartColumns[i], Length = length, SpaceLength = 0 });
+
+            }
+            return ColumnInfoList;
+        }
+
+        // TODO: still not right

```

Comment:
Can you specify what isn't right?

Reply:
it's mostly right now - another missed cleanup